lib/cli/src/commands/run/wasi.rs

```diff
@@ -105,6 +107,34 @@ pub struct Wasi {
     #[clap(long = "enable-async-threads")]
     pub enable_async_threads: bool,
 
+    /// Specifies the snapshot file that Wasmer will use to store
+    /// the state of the WASM process so that it can be later restored
+    #[cfg(feature = "snapshooter")]
+    #[clap(long = "snapshot-to")]
+    pub snapshot_to: Option<PathBuf>,
+
+    /// Indicates what events will cause a snapshot to be taken
+    /// and written to the snapshot file.
+    ///
+    /// If not specified, the default is to snapshot on idle plus if a
+    /// snapshot period is provided it will also default to periodic snapshots
+    /// as well.
+    #[cfg(feature = "snapshooter")]
+    #[clap(long = "snapshot-on")]
+    pub snapshot_on: Vec<SnapshotTrigger>,
+
+    /// Time in seconds between taking snapshots of the process and dumping
+    /// them to the snapshot file.
+    #[cfg(feature = "snapshooter")]
+    #[clap(long = "snapshot-period")]
+    pub snapshot_period: Option<u64>,
```

Comment:
Why not making the dumping instant? (why would be wanted to be delayed?)

Reply:
It is actually instant already however this parameter allows for a timer trigger to be added and for the interval to be set. The idea of a timer trigger is so that those that want a real-time backup of a process that can be restored somewhere else can use this functionality.

Renamed it from `period` to `timer` to make it less ambiguous
https://github.com/wasmerio/wasmer/pull/4263/commits/8f11ac78b00d3f07b774e776d20e54221dc9e33f